File: models/blog_post.py
import re
from . import db


def slugify(text):
    text = text.lower().strip()
    text = re.sub(r'[^\w\s-]', '', text)
    text = re.sub(r'[-\s]+', '-', text)
    return text[:80]


class BlogPost(db.Model):
    __tablename__ = 'blog_posts'

    id = db.Column(db.Integer, primary_key=True)
    titulo = db.Column(db.String(200), nullable=False)
    slug = db.Column(db.String(80), unique=True, nullable=False, index=True)
    conteudo = db.Column(db.Text, nullable=False)
    resumo = db.Column(db.String(300))
    autor_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    categoria = db.Column(db.String(50), default='geral')
    imagem_url = db.Column(db.String(500))
    publicado = db.Column(db.Boolean, default=False)
    data_criacao = db.Column(db.DateTime, default=db.func.now())
    data_publicacao = db.Column(db.DateTime)

    autor = db.relationship('User', backref='posts')

    def save(self, commit=True):
        if not self.slug:
            base = slugify(self.titulo)
            slug = base
            counter = 1
            while BlogPost.query.filter_by(slug=slug).first():
                slug = f'{base}-{counter}'
                counter += 1
            self.slug = slug
        db.session.add(self)
        if commit:
            db.session.commit()

File: src/main/java/com/conexoessolidarias/model/BlogPost.java
package com.conexoessolidarias.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blog_posts")
public class BlogPost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(unique = true, nullable = false, length = 80)
    private String slug;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String conteudo;

    @Column(length = 300)
    private String resumo;

    @ManyToOne
    @JoinColumn(name = "autor_id", nullable = false)
    private User autor;

    private String categoria = "geral";

    @Column(length = 500)
    private String imagemUrl;

    @Column(nullable = false)
    private Boolean publicado = false;

    private LocalDateTime dataCriacao = LocalDateTime.now();

    private LocalDateTime dataPublicacao;

    public BlogPost() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }
    public String getSlug() { return slug; }
    public void setSlug(String slug) { this.slug = slug; }
    public String getConteudo() { return conteudo; }
    public void setConteudo(String conteudo) { this.conteudo = conteudo; }
    public String getResumo() { return resumo; }
    public void setResumo(String resumo) { this.resumo = resumo; }
    public User getAutor() { return autor; }
    public void setAutor(User autor) { this.autor = autor; }
    public String getCategoria() { return categoria; }
    public void setCategoria(String categoria) { this.categoria = categoria; }
    public String getImagemUrl() { return imagemUrl; }
    public void setImagemUrl(String imagemUrl) { this.imagemUrl = imagemUrl; }
    public Boolean getPublicado() { return publicado; }
    public void setPublicado(Boolean publicado) { this.publicado = publicado; }
    public LocalDateTime getDataCriacao() { return dataCriacao; }
    public void setDataCriacao(LocalDateTime dataCriacao) { this.dataCriacao = dataCriacao; }
    public LocalDateTime getDataPublicacao() { return dataPublicacao; }
    public void setDataPublicacao(LocalDateTime dataPublicacao) { this.dataPublicacao = dataPublicacao; }
}
